test_fms/topography/test_topography.F90

```diff
@@ -0,0 +1,310 @@
+
+    call write_data(top_fileobj, "ipts",   real(ipts, TEST_TOP_KIND_))
+    call write_data(top_fileobj, "jpts",   real(jpts, TEST_TOP_KIND_))
+    call write_data(top_fileobj, "xdat",   xdat)
+    call write_data(top_fileobj, "ydat",   ydat)
+    call write_data(top_fileobj, "zdat",   zdat)
+
+    call close_file(top_fileobj)
+
+  else
+    call mpp_error(FATAL, "test_topography: error opening topog_file")
+  end if
+  !-------------------------------------------------------------------------------------------------------------!
+
+  ! write water file
+  if (open_file(top_fileobj, water_file, "overwrite")) then
+    call register_axis(top_fileobj, "i_zdat", ipts)   !first index dimension in zdat
+    call register_axis(top_fileobj, "j_zdat", jpts)   !second index dimension in zdat
+    call register_axis(top_fileobj, "ipts_r", ipts_r)
+    call register_axis(top_fileobj, "jpts_r", jpts_r)
+    call register_axis(top_fileobj, "i_xdat", iptsp1) !# of points in xdat variable
+    call register_axis(top_fileobj, "j_ydat", jptsp1) !# of point in ydat variable
+
+    call register_field(top_fileobj, "ipts",   "double", dimensions=(/"ipts_r"/))
+    call register_field(top_fileobj, "jpts",   "double", dimensions=(/"jpts_r"/))
+    call register_field(top_fileobj, "xdat",   "double", dimensions=(/"i_xdat"/))
+    call register_field(top_fileobj, "ydat",   "double", dimensions=(/"j_ydat"/))
+    call register_field(top_fileobj, "zdat",   "double", dimensions=(/"i_zdat", "j_zdat"/))
+
+    call write_data(top_fileobj, "ipts",   real(ipts, TEST_TOP_KIND_))
+    call write_data(top_fileobj, "jpts",   real(jpts, TEST_TOP_KIND_))
+    call write_data(top_fileobj, "xdat",   xdat)
+    call write_data(top_fileobj, "ydat",   ydat)
+    call write_data(top_fileobj, "zdat",   zdat)
+
+    call close_file(top_fileobj)
+
+  else
+    call mpp_error(FATAL, "test_topography: error opening water_file")
+  end if
+  !-------------------------------------------------------------------------------------------------------------!
+
+  call test_topog_mean     ; call test_topog_stdev
+  call test_get_ocean_frac ; call test_get_ocean_mask
+  call test_get_water_frac ;call test_get_water_mask
+
+  call fms_end
+
+  contains
+
+  subroutine test_topog_mean()
+
+    implicit none
+    real(kind=TEST_TOP_KIND_), dimension(size(lon2d,1)-1,size(lat2d,2)-1) :: zmean2d
+    real(kind=TEST_TOP_KIND_), dimension(size(lon1d)-1,size(lat1d)-1)     :: zmean1d
+    logical                                                               :: get_mean_answer
+
+    !---------------------------------------- test topog mean 2d ---------------------------------------------!
+
+    get_mean_answer = get_topog_mean(lon2d, lat2d, zmean2d)
+
+    if (get_mean_answer .neqv. .true.) call mpp_error(FATAL, "topog field not read correctly")
+    call check_answers(zmean2d(1,1), 0.5_lkind, "Error in test_topog_mean 2d")
```

Comment:
Should the other values of zmean2d be checked?

Reply:
added comments to make this more understandable in 4f73bb2d